### ISSUE TITLE
Slow server updates causing perpetual errors

### ISSUE THREAD

**Issue by berriganator** [HabitRPG/habitica#6610]
I am Firefox 43.0.4 on a Mac. Habitica has been really slow and frustrating to use for some time now, and it has gotten worse in recent weeks. I have to wait for the the blue server update bar to finish loading to do anything. If I start adding a checklist item or checking off a task in the middle of refresh, it will abort my current activity and I have to start over. It is most noticeable when I add a new task with a checklist. After adding a checklist item, the task will save and automatically collapse. If I am in the middle of typing a new checklist item, it will save the box but not the text I was writing in the box. So I have to click "edit" and type in a checklist item each time. The lag is really annoying and discourages me from adding tasks because it ends up taking so long.

**Comment by Alys** [comment]
I'm sorry about the delay in replying.

How many tasks do you have? The easiest way to count them is to go to Data -> Data Display Tool and use the "Task Statistics" section.

Do you have another browser installed on your Mac or are you able to install one? Is the site equally slow on it?

**Comment by berriganator** [comment]
Hello. Yes, I have the same issue in Chrome of the tasks collapsing prematurely before I'm even done typing the checklists, tagging or editing them in anyway. It happens as well across multiple computers. 

I have about 67 tasks. 

Thanks!

**Comment by Ant0ni013** [comment]
Hi, I have been experiencing the same thing with chrome 47.0.2526.83 on my galaxy s5 phone (Android 5.0.0; SM-G900I Build/LRX21T) in particular, and occasionally on my laptop, also using chrome (48.0.2564.97 m), windows 10.
A related issue has also caused a strange result: I ticked off a bunch of dailies on my phone, which caused me to level up to 10. At this stage the "pick your class" box appeared, as well as a (I think) 408 error. 
I dismissed the error and chose my class as mage. I then locked my phone without refreshing the page.
Later I opened up habitica again, this time on my laptop. The server had not received the info on all of the dailies, they were unchecked and I was still level 9. However, I was a mage, with mana, mage gear unlocked, and 9 skill points available to spend, all things that normally occur at lvl10.

![l9 mage](https://cloud.githubusercontent.com/assets/17004656/12716789/41ae2440-c937-11e5-85ac-59f71c49dda6.png)
.
I'm still lvl9 at time of posting, screenshot attached. I don't think the console is available on phones, so I can't provide a log.
I have 104 tasks.
User ID: 65ea127a-338e-4dae-92c7-6480a0e1b50b
